### PR TITLE
support ci train mode and grad comparison

### PR DESCRIPTION
**Pull request by xiaoyao0115**
### PR Category

CINN

### PR Types

Not User Facing

### Description

support ci train mode and grad comparison

### CODE REVIEW

File: test/ir/pir/cinn/sub_graphs/base.py [hunk 1]
```diff
@@ -77,3 +90,22 @@ def test_ast_prim_cinn(self):
                 np.testing.assert_allclose(
                     st.numpy(), cinn.numpy(), atol=self.atol
                 )
+        if self.with_train:
+            criterion = nn.MSELoss()
+            target = paddle.rand(shape=st_out.shape, dtype=st_out.dtype)
```
Comment (Aurelius84) [inline test/ir/pir/cinn/sub_graphs/base.py:95]: 更简单的，其实可以对输出直接求mean，然后loss.backward()

File: test/ir/pir/cinn/sub_graphs/test_sub_graph_0.py [hunk 2]
```diff
@@ -135,6 +135,7 @@ def init(self):
             paddle.rand(shape=[22, 512, 7, 7], dtype=paddle.float32),
         )
         self.net = LayerCase()
```
Comment (Aurelius84) [inline test/ir/pir/cinn/sub_graphs/test_sub_graph_0.py:137]: 此处是传递的net 实例，也就意味着cinn和动转静模式使用的是同一份，这里是有风险的，更建议考虑将此处传递为类，在TestBase里分别在同样的随机种子seed下实例化。

目前之所以没有问题，是因为只是求了梯度，但没有进行优化器更新参数，所以复用上没有问题。但在包含Norm类算子时，即使不更新参数，也可能有潜在风险。

**Review by Aurelius84 — APPROVED** [review]
LGTM
